### PR TITLE
Ingest MRI data

### PR DESCRIPTION
**Pull request by mklarqvist**
Ingest data from DICOMs and store as
* structured data as a partitioned parquet file in hive layout
* HDF5 file with each HWC tensor compressed as a blob

This approach is around 7-fold smaller compared to our previous approach with uncompressed float32 HW tensors stored in C different datasets

Related to #405

### CODE REVIEW

**Comment by mklarqvist** [comment]
Example layout for the DICOM meta data
```
body_mri_manifest/
├── _common_metadata
├── _metadata
└── ukbid=<hidden>
    └── part.0.parquet
└── ukbid=<hidden>
    └── part.0.parquet
...
```

File: ml4h/applications/ingest/ingest_mri.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
+import pydicom as dicom
+from dicom_parser import Image
+from dicom_parser.utils.siemens.csa.header import CsaHeader
+import numpy as np
+import pandas as pd
+import glob
+import fastparquet as fp
+import blosc
+import xxhash
```
Comment (ndiamant) [inline ml4h/applications/ingest/ingest_mri.py:9]: Should we add these to our requirements?

File: ml4h/applications/ingest/ingest_mri.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+sample_manifest['Slice Thickness'] = sample_manifest['Slice Thickness'].astype(np.float32)
+sample_manifest['Window Width'] = sample_manifest['Window Width'].astype(np.float32)
+sample_manifest['dB/dt'] = sample_manifest['dB/dt'].astype(np.float32)
+
+# Reformat column names to be all lower case and replace spaces with underscores
+sample_manifest.columns = [s.lower().replace(' ','_') for s in sample_manifest.columns.values]
+
+
+# Store meta data
+fp.write('body_mri_manifest',
```
Comment (ndiamant) [inline ml4h/applications/ingest/ingest_mri.py:166]: If we want to parallelize will appending be a problem?

**Review by ndiamant — COMMENTED** [review]
I have a few questions about how we will run this script
- Are we going to parallelize it?
    - If so, how does writing to the parquet file work?
    - If not, will the execution be too slow?
- How do you want to integrate with the existing ingest code?
- Can you write a simple `tensor_from_file` helper function as a demonstration of loading the compressed data ?

File: ml4h/applications/ingest/ingest_mri.py (hunk 1 above)
Comment (mklarqvist) [reply ml4h/applications/ingest/ingest_mri.py:9]: I don't believe that is appropriate in this case. The core functionality of ML4H has no dependency of these. These are only required in the special case when a user require the ingestion of MRI data from the UKBioBank.

File: ml4h/applications/ingest/ingest_mri.py (hunk 2 above)
Comment (mklarqvist) [reply ml4h/applications/ingest/ingest_mri.py:166]: A 2-pass would be appropriate in the embarrassingly parallel case. We could modify this to saving individual Parquet files in `simple` mode --- when all these are finished we could merge them as above in `hive` mode.

Comment (mklarqvist) [reply ml4h/applications/ingest/ingest_mri.py]: It's now in simple mode. A second pass will be required to reduce the map.

**Pull request closed without merging**